ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java

```diff
@@ -111,19 +111,37 @@ private static Executable createExecutable(
 
   @VisibleForTesting
   static void enforceStreamStateDirAvailability(final File streamsStateDir) {
-    if (!streamsStateDir.exists() || !streamsStateDir.isDirectory()) {
-      throw new KsqlServerException("The kafka streams state directory does not exist: "
-          + streamsStateDir.getPath()
-          + "\n Make sure the directory exists and is writable for KSQL server."
+    if (!streamsStateDir.exists()) {
+      final boolean mkDirSuccess = streamsStateDir.mkdir();
```

Comment:
Use `mkdirs` here. This way the path up to this directory will be created (this is also what streams does).

Reply:
Changed it to `mkdirs`.